specification/quota/resource-manager/Microsoft.Quota/preview/2021-03-15-preview/quota.json

```diff
@@ -655,9 +755,113 @@
           "description": "Quota request ID.",
           "type": "string",
           "readOnly": true
+        },
+        "limit": {
+          "description": "Resource quota limit properties.",
+          "$ref": "#/definitions/LimitJsonObject"
         }
       }
     },
+    "LimitObjectTypes": {
+      "description": "The limit object type.",
+      "enum": [
+        "LimitValue",
+        "UsagesValue"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "limitType",
+        "modelAsString": true
+      }
+    },
+    "LimitTypes": {
+      "description": "The quota or usages limit types.",
+      "enum": [
+        "Independent",
+        "Shared"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "quotaLimitTypes",
+        "modelAsString": true
+      }
+    },
+    "LimitJsonObject": {
```

Comment:
I don't think you need polymorphism between the usages and quota APIs since they are separate APIs that serve different needs. I thought from discussions you just wanted some level of polymorphism within each API to represent resource type specific properties

Reply:
I will remove the polymorphism between usages and quota API.
Leave it for quota/limit, so that we can add addition properties, when MG quota is onboarded.

Reply:
Separated quotas and usages from  the polymorphism .